internal/locate/cse.go

```diff
@@ -0,0 +1,129 @@
+package locate
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/pkg/errors"
+	"github.com/tikv/client-go/v2/util/codec"
+	pd "github.com/tikv/pd/client"
+)
+
+var _ pd.Client = &CSEClient{}
+
+type CSEClient struct {
+	pd.Client
+
+	mu struct {
+		sync.Mutex
+		stores []*metapb.Store
+	}
+
+	httpClient *http.Client
+}
+
+func NewCSEClient(ctx context.Context, client pd.Client) (*CSEClient, error) {
+	stores, err := client.GetAllStores(ctx)
```

Comment:
stores maybe change

Reply:
> stores maybe change

This version of the code is intended to be a POC, it will be refactored while the CSE region interface is stable.